distributed/worker.py

```diff
@@ -3359,11 +3359,10 @@ async def _maybe_deserialize_task(
             logger.error("Could not deserialize task", exc_info=True)
             self.log.append((ts.key, "deserialize-error", stimulus_id, time()))
             emsg = error_message(e)
-            emsg.pop("status")
             self.transition(
                 ts,
                 "error",
-                **emsg,
+                **{k: v for k, v in emsg.items() if k != "status"},
```

Comment:
I considered the previous version actually better readable. Is there a reason for this change?